android-json-form-wizard/src/main/java/com/vijay/jsonwizard/activities/JsonWizardFormActivity.java

```diff
@@ -8,7 +8,7 @@
 /**
  * Created by keyman on 04/12/2018.
  */
-public class JsonWizardFormActivity extends JsonFormActivity {
+public class JsonWizardFormActivity extends FormConfigurationJsonFormActivity {
```

Comment:
Why not add this to `JsonFormBaseActivity` or `JsonFormActivity` to support configurability of all kinds of forms and not just wizard forms?

Reply:
Good idea

Reply:
This extends the `JsonFormActivity` and `JsonWizardFormActivity` extends `JsonFormActivity` activity too. `FormConfigurationJsonFormActivity` can't extend `JsonFormBaseActivity` without a lot of refactor as it depends on a couple of functions on `JsonFormActivity`.

Reply:
Cool, but doesn't that still mean that form configurability with not be supported for `JsonFormActivity`?

Reply:
Yes, [this](https://smartregister.atlassian.net/wiki/spaces/Documentation/pages/1507000340/OpenSRP+Document+Configurability) upgrade doc explains why everyone should now extend `FormConfigurationJsonFormActivity` instead of `JsonFormActivity`

Reply:
Ok, I guess it's a bit asymmetric that we have `JsonWizardFormActivity` directly inheriting from `FormConfigurationJsonFormActivity` rather than creating a separate `FormConfigurationJsonWizardFormActivity` and asking people using wizard forms to migrate to that as well.

Reply:
It does make sense that `FormConfigurationJsonFormActivity` is a child of `JsonFormActivity` i.e. semantically it's a version of `JsonFormActivity`.

Reply:
On the other hand, it feels like all `JsonFormActivity`s should have the configurable feature available to them but not enabled by default. And so it would make more sense if it was added in the parent class and inherited by the child classes. Not sure if this indicates a use case for composition over inheritance here.

Reply:
Anyway, I don't see any functional issues with the approach. But I think it's worth pointing out the assymmetry. And noting that now we consider `JsonWizardFormActivity` to be a type of `FormConfigurationJsonFormActivity` and not a type of `JsonFormActivity` directly. Meaning that you will always have the configurability feature available (is it mandatory?) for wizards.

Cool, are the changes tested for compatibility

Reply:
Anyway, I don't see any functional issues with the approach. But I think it's worth pointing out the asymmetry. And noting that now we consider `JsonWizardFormActivity` to be a type of `FormConfigurationJsonFormActivity` and not a type of `JsonFormActivity` directly. Meaning that you will always have the configurability feature available (is it mandatory?) for wizards.

Cool, are the changes tested for compatibility

Reply:
Anyway, I don't see any functional issues with the approach. But I think it's worth pointing out the asymmetry. And noting that now we consider `JsonWizardFormActivity` to be a type of `FormConfigurationJsonFormActivity` and not directly a type of `JsonFormActivity`. Meaning that you will always have the configurability feature available (is it mandatory?) for wizards.

Reply:
Are the changes tested for compatibility?